library/core/src/num/int_macros.rs

```diff
@@ -3,10 +3,10 @@ macro_rules! int_impl {
         Self = $SelfT:ty,
         ActualT = $ActualT:ident,
         UnsignedT = $UnsignedT:ty,
-        BITS = $BITS:expr,
-        BITS_MINUS_ONE = $BITS_MINUS_ONE:expr,
-        Min = $Min:expr,
-        Max = $Max:expr,
+        BITS = $BITS:literal,
+        BITS_MINUS_ONE = $BITS_MINUS_ONE:literal,
+        Min = $Min:literal,
+        Max = $Max:literal,
         rot = $rot:expr,
         rot_op = $rot_op:expr,
```

Comment:
All of these should probably also be literals (also `rot` should be passed as a string literal instead of a number literal as well, I only see it being used in docs)

Reply:
I changed the non-comment uses of `$BITS` to `Self::BITS`, and moved everything to `literal` except the things passed as macros.

@rustbot ready